src/lib/builders/table-of-contents/types.ts

```diff
@@ -0,0 +1,46 @@
+
+export type Heading = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+/**
+ * The ToC type describes which headings should be considered active.
+ */
+export type TOCType = 'lowest' | 'highest' | 'all-active' | 'lowest-parents' | 'highest-parents';
+
+/**
+ * Arguments to be passed to the Table of Contents builder.
+ * 
+ * @param exclude - A list of headings that should be excluded in the table.
+ * @param scrollOffset - The pixel offset that should be added when scrolling to a heading.
+ * @param scrollBehaviour - Describes whether the scroll behaviour should be smooth or not.
+ * @param tocType - Describes which headings should be considered active.
+ */
+export type CreateTableOfContentsArgs = {
+    selector: string,
+    exclude?: Heading[];
+    scrollOffset?: number;
+    scrollBehaviour?: 'smooth' | 'jump';
+    tocType?: TOCType;
```

Comment:
Maybe just type?

Reply:
Yes makes sense. I'll update it.

Reply:
I decided to name it `activeType` to be more precise about what it is ^^'